### ISSUE TITLE
Support Markdown in name to produce text+HTML names

### ISSUE THREAD

**Issue by foolip** [web-platform-dx/web-features#930]
https://github.com/web-platform-dx/web-features/pull/935 shows what this would look like.

**Comment by foolip** [comment]
There's a bunch of good discussion in https://github.com/web-platform-dx/web-features/pull/935, which I've now abandoned.

The most important thing is that nobody seems to really want `<code>` in feature names, but @captainbrosset would like a field with the name escaped as HTML.

**Comment by captainbrosset** [comment]
> but @captainbrosset would like a field with the name escaped as HTML.

Yes, I think it'd be good if consumers (including myself) had access to a name that's safe to put in HTML without having to worry about it.

**Comment by foolip** [comment]
Providing that would be easy, and given the lack of interest (including my own) for marking up names with `<code>` I guess we can simply call it `name_html` and revisit later if we want more variants.

**Comment by ddbeck** [comment]
See also https://github.com/web-platform-dx/web-features/issues/2018 for linting names.

**Comment by tidoust** [comment]
Discussed in [today's web-features meeting](https://docs.google.com/document/d/1ree75ImLZjf60lTZ3BhCaLHygxgywr7SBXp-q0xPs8A/edit?tab=t.6r13tylx2c9k#heading=h.r0t8r6q9buvs). The need has not materialized in practice. This could be revisited in the future.

I don't think that we discussed the possibility to provide a name that is safe for inclusion in HTML, though?
